core/lib/Thelia/Action/Product.php

```diff
@@ -100,6 +121,451 @@ public function create(ProductCreateEvent $event)
+
+                    // Create and dispatch event
+                    $cpCreateFileEvent = new FileCreateOrUpdateEvent($cp->getId());
+                    $cpCreateFileEvent
+                        ->setModel($cpFile)
+                        ->setUploadedFile($cpCopiedFile)
+                        ->setParentName($cp->getTitle());
+
+                    switch ($type) {
+                        case 'images':
+                            $dispatcher->dispatch(TheliaEvents::IMAGE_SAVE, $cpCreateFileEvent);
+
+                            // Get original product image I18n
+                            $opFileI18ns = ProductImageI18nQuery::create()
+                                ->findById($opFile->getId());
+                            break;
+
+                        case 'documents':
+                            $dispatcher->dispatch(TheliaEvents::DOCUMENT_SAVE, $cpCreateFileEvent);
+
+                            // Get original product document I18n
+                            $opFileI18ns = ProductDocumentI18nQuery::create()
+                                ->findById($opFile->getId());
+                            break;
+                    }
+
+                    // Set temporary source file as original one
+                    $fs->rename($srcTmp, $srcPath);
+
+                    // Set clone files I18n
+                    foreach ($opFileI18ns as $opFileI18n) {
+                        // Update file with current I18n info. Update or create I18n according to existing or absent Locale in DB
+                        $cpFile
+                            ->setLocale($opFileI18n->getLocale())
+                            ->setTitle($opFileI18n->getTitle())
+                            ->setDescription($opFileI18n->getDescription())
+                            ->setChapo($opFileI18n->getChapo())
+                            ->setPostscriptum($opFileI18n->getPostscriptum());
+
+                        // Create and dispatch event
+                        $cpUpdateFileEvent = new FileCreateOrUpdateEvent($cp->getId());
+                        $cpUpdateFileEvent->setModel($cpFile);
+
+                        switch ($type) {
+                            case 'images':
+                                $dispatcher->dispatch(TheliaEvents::IMAGE_UPDATE, $cpUpdateFileEvent);
+                                break;
+
+                            case 'documents':
+                                $dispatcher->dispatch(TheliaEvents::DOCUMENT_UPDATE, $cpUpdateFileEvent);
+                                break;
+                        }
+                    }
+                } else {
+                    Tlog::getInstance()->addWarning("Failed to find media file $srcPath");
+                }
+            }
+        }
+    }
+
+    public function createClonePSE(ProductModel $cp, $opPSE, $currencyId, EventDispatcherInterface $dispatcher)
+    {
```

Comment:
Management of PSE cloning shoud not be defined in the Product action, but in the ProductSaleElement action, and invoked by proper events.